scripting/FileParsing.js

```diff
@@ -1 +1,141 @@
+              return jsonObj;
+          }
+          else{
+              //yeild error because wrong file type
+              alert("incorrect file type: please enter a JSON or CVS file and try again.");
+          }
+  }
+  //method to replace characters in a string based upon index
+  String.prototype.replaceAt = function(index, replacement) {
+  	if (index >= this.length) {
+  		return this.valueOf();
+  	}
+
+  	return this.substring(0, index) + replacement + this.substring(index + 1);
+  }
+
+  function csvHandeling(array){
+      //needs to clean extra commas to keep data together
+      //there are commas within cells so it switches them with "|" to avoid splitting the cells
+      let isComma = false;
+      for(y = 0; y < array.length; y++){
+          for(x = 0; x < array[y].length; x++){
+              if(array[y].charAt(x) == "\""  && !isComma){
+                  isComma = true;
+              }
+              else if(array[y].charAt(x) == "\""  && isComma){
+                  isComma = false;
+              }
+              if(isComma && array[y].charAt(x) == ","){
+                  array[y] = array[y].replaceAt(x,"^");
+              }
+          }
+      }
+      //breaks array into multidimentional array - essentially a grid same as the excel doc
+      for(let x = 0; x < array.length; x++){
+          array[x] = array[x].split(",");
+      }
+      //finds which column has mod_id and deletes all the garbage before it, deletes everything between mod_id and AM, deletes everything after the last loci
+      let correctColumn = 0;
+      while(array[0][correctColumn] != "mod_id"){
+          correctColumn ++;
+      }
+      for(let y = 0; y < array.length; y++){
+          array[y].splice(0,correctColumn);
+      }
+      correctColumn = 0;
+      while(array[0][correctColumn] != "AM"){
+          correctColumn ++;
+      }
+
+      for(let y = 0; y < array.length; y++){
+          array[y].splice(1,correctColumn - 1);
+      }
+      correctColumn = 0;
+      for(x = 0; x < array[0].length; x++){
+          if(array[0][x].slice(0,3) == "mod" && array[0][x] != "mod_id"){
+              correctColumn = x;
+          }
+      }
+      if(correctColumn > 0){
+          for(let y = 0; y < array.length; y++){
+              array[y].splice(correctColumn,array[0].length - 1);
+          }
+      }
+
+      for(y = 0; y < array.length; y++){
+          for(x = 0; x < array[y].length; x++){
+              for(z = 0; z < array[y][x].length; z++){
+                  if(array[y][x].charAt(z) == "\""){
+                      array[y][x] = array[y][x].substr(1);
+                      array[y][x] = array[y][x].substr(0, array[y][x].length - 1);
+                  }
+              }
+          }
+      }
+      for(y = 0; y < array.length; y++){
+          for(x = 1; x < array[y].length; x++){
+              array[y][x] = array[y][x].split("^");
+          }
+      }
+      //object maker
+      let objArray = [];
```

Comment:
Again, this function seems very long, might want to split this object maker into another function